### PR TITLE
ENH: Add ``layout_mode_font_height_weight`` argument to ``PageObject.extract_text()``

### PR DESCRIPTION
**Pull request by hpierre001**
Implements feature proposed in issue #2915

### CODE REVIEW

File: tests/test_text_extraction.py [hunk 1]
```diff
@@ -219,3 +219,13 @@ def test_text_leading_height_unit():
     page = reader.pages[0]
     extracted = page.extract_text()
     assert "Something[cited]\n" in extracted
+
+
+def test_layout_mode_vertical_space():
+    """Tests for #2915"""
+    reader = PdfReader(RESOURCE_ROOT / "attachment.pdf")
```
Comment (stefan6419846) [inline tests/test_text_extraction.py:226]: I would recommend using `crazyones.pdf` instead which should have the same content.

Apart from this: Could you please extend the test to actually show the difference between different weights? By just looking at the existing test, it is not really clear to me where the actual difference is (unless looking at others tests or `crazyones.txt`).

File: tests/test_text_extraction.py [hunk 2]
```diff
@@ -219,3 +219,44 @@ def test_text_leading_height_unit():
...
+
+        text = page.extract_text(extraction_mode="layout", layout_mode_space_vertically=True).encode("utf-8")
+
+        # Compare the text of the PDF to a known source
+        for expected_line, actual_line in zip(text.splitlines(), pdftext.splitlines()):
+            assert expected_line == actual_line
+
+        pdftext = pdftext.replace(b"\r\n", b"\n")  # fix for windows
+        assert text == pdftext, (
+            "PDF extracted text differs from expected value.\n\n"
```
Comment (stefan6419846) [inline tests/test_text_extraction.py:243]: Doesn't *pytest* already provide the diff accordingly?

Comment (hpierre001) [reply tests/test_text_extraction.py:243]: Maybe, i just copied the test using crazyones.
I'll look into it and maybe update the other test too.

Comment (stefan6419846) [reply tests/test_text_extraction.py:243]: Thanks for the hint. I have not been aware of this.

**Review by stefan6419846 — APPROVED** [review]

**Comment by stefan6419846** [comment]
Thanks for your contribution.

**Pull request merged**